### PR TITLE
use a lazy user for the file owner when listing a directory

### PR DESCRIPTION
**Pull request by icewind1991**
pretty much only `getUID` and `getDisplayName` are called on the file owner objects anyway

### CODE REVIEW

File: lib/private/User/LazyUser.php [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+
+	public function getQuota() {
+		return $this->getUser()->getQuota();
+	}
+
+	public function setQuota($quota) {
+		$this->getUser()->setQuota($quota);
+	}
+
+
```
Comment (CarlSchwan) [inline lib/private/User/LazyUser.php:148]:
```suggestion
```

**Review by CarlSchwan — APPROVED** [review]
Nice solution and good that the displayNameCache can be reused :)

**Review by juliusknorr — APPROVED** [review]
Nice :)

**Pull request merged**